lib/slack-rtmapi/client.rb

```diff
@@ -81,14 +81,22 @@ def connected?
       @connected || false
     end
 
+    def read_socket
+      @reading = true
+      data = @socket.readpartial 4096
+      @driver.parse data if !data.nil? && !data.empty?
+      @reading = false
+    end
+
     # All the polling work is done here
     def inner_loop
       return if @stop
-      data = @socket.readpartial 4096
-      return if data.nil? or data.empty?
-
-      @driver.parse data
-      @msg_queue.each {|msg| @driver.text msg}
+      unless @reading
+        Thread.new do
```

Comment:
Sorry, I my policy will be strict here:
- no Thread
- no EventMachine
- no Celluloid
- no Rack
- no whatever

The client is only the bare metal, and it's the wrappers which will expose the threaded version.
I can code an example draft soon, but keep in mind that threads have their own issues.

We will not force the lib user to use Thread if he don't want (and many don't want thread).

Please let the inner loop as simple as possible, without threads.